### PR TITLE
Implemented Check for Configurable Products

### PR DESCRIPTION
**Pull request by xapiksw**

### CODE REVIEW

File: src/Validator/Product.php [hunk 1]
```diff
@@ -41,9 +59,47 @@ public function validateRequest(RequestInterface $request): bool
         $urlKey = $this->getPathFrontName($request);
         $productCollection = $this->productCollection->clear();
         $productCollection->addAttributeToFilter('url_key', $urlKey);
+
         $ids = $productCollection->getAllIds();
         $productId = reset($ids);
 
-        return !!$productId;
+        if (!$productId) return false;
```
Comment (eli-l) [inline src/Validator/Product.php]: Please organize the code accordingly to PSR-2 https://www.php-fig.org/psr/psr-2/

File: src/Validator/Product.php [hunk 2]
```diff
@@ -41,9 +59,47 @@ public function validateRequest(RequestInterface $request): bool
...
+
+        $typeIds = $productCollection->getProductTypeIds();
+        $type = reset($typeIds);
+        $parameters = $request->getParams();
+
+        if ($type === Configurable::TYPE_CODE && !empty($parameters) && !$this->checkConfigurableProduct($productId, $parameters))
```
Comment (eli-l) [inline src/Validator/Product.php]: Might be worth to split to multi-lines for a better readability

Comment (eli-l) [reply src/Validator/Product.php]: Or even move _!empty($parameters)_ to the early return

File: src/Validator/Product.php [hunk 3]
```diff
@@ -41,9 +59,47 @@ public function validateRequest(RequestInterface $request): bool
...
+
+    /**
+     * Checks whether configurable product with specified parameters does exist
+     * @param int $productId
+     * @param array $parameters
+     * @return bool
+     */
+    protected function checkConfigurableProduct(int $productId, array $parameters): bool {
+        $product = $this->product->load($productId);
+
```
Comment (eli-l) [inline src/Validator/Product.php:92]: what's the idea behind this empty line?

File: src/Validator/Product.php [hunk 4]
```diff
@@ -41,9 +59,47 @@ public function validateRequest(RequestInterface $request): bool
...
+        foreach ($attributes as $attribute) {
+            $attributeCode = $attribute->getProductAttribute()->getAttributeCode();
+
+            if (array_key_exists($attributeCode, $parameters)) {
+                $parameterValue = $parameters[$attributeCode];
+                unset($parameters[$attributeCode]);
+                $options = $attribute->getOptions();
+
+                if (array_search($parameterValue, array_column($options,  'value_index')) === false) return false;
```
Comment (eli-l) [inline src/Validator/Product.php]: Please format it accordingly

File: src/Validator/Product.php [hunk 5]
```diff
@@ -41,9 +59,47 @@ public function validateRequest(RequestInterface $request): bool
...
+            }
+        }
+
+        if (!empty($parameters)) return false;
```
Comment (eli-l) [inline src/Validator/Product.php]: this check is already in a call stack, and a method can be called in current class or subclass, looks like this line could be easily removed.

**Review by eli-l — CHANGES_REQUESTED** [review]
OK in general, please pay attention to your code's readability (following PSRs, organizing logic in an easier manner).

File: src/Validator/Product.php (hunk 5 above)
Comment (xapiksw) [reply src/Validator/Product.php]: Please, could you elaborate a bit more on it. The first check in `validateRequest` function prevents executing unnecessary logic execution when there are no parameters passed, as this one, paired with unset, ensures that no other query parameters despite the ones that are used by configurable product were passed.

**Pull request merged**